src/blocks/calendar.rs

```diff
@@ -0,0 +1,610 @@
+            }) => {
+                let credentials = if let Some(path) = credentials_path {
+                    util::deserialize_toml_file(path.expand()?.to_string())
+                        .error("Failed to read basic credentials file")?
+                } else {
+                    credentials.clone()
+                };
+                let BasicCredentials {
+                    username: Some(username),
+                    password: Some(password),
+                } = credentials
+                else {
+                    return Err(Error::new("Basic credentials are not configured"));
+                };
+                auth::Auth::basic(username, password)
+            }
+            AuthConfig::OAuth2(oauth2) => {
+                let credentials = if let Some(path) = &oauth2.credentials_path {
+                    util::deserialize_toml_file(path.expand()?.to_string())
+                        .error("Failed to read oauth2 credentials file")?
+                } else {
+                    oauth2.credentials.clone()
+                };
+                let OAuth2Credentials {
+                    client_id: Some(client_id),
+                    client_secret: Some(client_secret),
+                } = credentials
+                else {
+                    return Err(Error::new("Oauth2 credentials are not configured"));
+                };
+                let auth_url =
+                    AuthUrl::new(oauth2.auth_url.clone()).error("Invalid authorization url")?;
+                let token_url =
+                    TokenUrl::new(oauth2.token_url.clone()).error("Invalid token url")?;
+
+                let flow = OAuth2Flow::new(
+                    ClientId::new(client_id),
+                    ClientSecret::new(client_secret),
+                    auth_url,
+                    token_url,
+                    oauth2.redirect_port,
+                );
+                let token_store =
+                    TokenStore::new(Path::new(&oauth2.auth_token.expand()?.to_string()));
+                auth::Auth::oauth2(flow, token_store, oauth2.scopes.clone())
+            }
+        };
+        Ok(Self {
+            client: Client::new(
+                Url::parse(&config.url).error("Invalid CalDav server url")?,
+                auth,
+            ),
+            config,
+        })
+    }
+
+    async fn get_next_events(
+        &mut self,
+        within: Duration,
+    ) -> Result<OverlappingEvents, CalendarError> {
+        let calendars: Vec<_> = self
+            .client
+            .calendars()
+            .await?
+            .into_iter()
+            .filter(|c| self.config.calendars.is_empty() || self.config.calendars.contains(&c.name))
+            .collect();
+        let mut events: Vec<Event> = vec![];
+        for calendar in calendars {
+            let calendar_events: Vec<_> = self
+                .client
+                .events(
+                    &calendar,
+                    Utc::now()

```

Comment:
I finally was able to chase down why my events weren't showing up. This should be `Local::now()` instead of `Utc::now()` so that it's the local date part that were using.